pkgs/by-name/no/nocodb/package.nix

```
@@ -0,0 +1,100 @@
+{
+  lib,
+  stdenv,
+  nodePackages,
+  pnpm,
+  sqlite,
+  pkg-config,
+  rsync,
+  makeWrapper,
+  node-gyp,
+  vips,
+  fetchFromGitHub,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nocodb";
+  version = "0.260.2";
+
+  src = fetchFromGitHub {
+    owner = "nocodb";
+    repo = "nocodb";
+    tag = finalAttrs.version;
+    hash = "sha256-PoEH+HZBICUWDR3o5BKKPHWKsVWhBQxB+4dwFT8TdJA=";
+  };
+
+  buildPhase = ''
+    export NODE_OPTIONS="--max_old_space_size=16384"
+    export NUXT_TELEMETRY_DISABLED=1
+    export npm_config_nodedir=${nodePackages.nodejs}
+
+    pnpm --filter=nocodb-sdk run build
+    pnpm run registerIntegrations
+    pnpm --filter=nc-gui run build:copy
+    pnpm --filter=nocodb run docker:build
+  '';
+
+  installPhase = ''
+    mkdir -p $out/share/nocodb/packages/nocodb
+    cp -v ./packages/nocodb/docker/main.js $out/share/nocodb/packages/nocodb/index.js
+
+    # only ship nocodb workspace prod deps with node_modules (1.9GB -> 400MB)
+    rm -rf ./node_modules ./packages/nocodb/node_modules
+    pnpm install \
+        --offline \
+        --prod \
+        --ignore-scripts \
+        --filter=nocodb \
+        --frozen-lockfile
+    # nodejs 22.11.0 -> 22.12.0 broke pnpm rebuild somehow, so let's do it manaully
+    # pnpm rebuild -r --verbose --reporter=append-only
+    for package in $(find -L packages/nocodb/node_modules -name binding.gyp -type f); do
+        cd "$(dirname "$package")"
+        node-gyp rebuild
+        cd -
```

```suggestion
      --offline \
      --prod \
      --ignore-scripts \
      --filter=nocodb \
      --frozen-lockfile
    # nodejs 22.11.0 -> 22.12.0 broke pnpm rebuild somehow, so let's do it manaully
    # pnpm rebuild -r --verbose --reporter=append-only
    for package in $(find -L packages/nocodb/node_modules -name binding.gyp -type f); do
      cd "$(dirname "$package")"
      node-gyp rebuild
      cd -
```
we use two spaces in nix files also for shell code